### ISSUE TITLE
Is the network really well trained?

### ISSUE THREAD

**Issue by EthanZhangYi** [wasidennis/AdaptSegNet#29]
I try to run the code with `pytorch0.4.0`, and the result of `resnet101-multi-level` is reproduced, but it is very **unstable**. Only at iteration 105000, the result in paper (42.4%) is reproduced.
```
iteration 5000      0.3098
iteration 10000     0.3488
iteration 15000     0.3575
iteration 20000     0.3531
iteration 25000     0.3715
iteration 30000     0.3851
iteration 35000     0.4022
iteration 40000     0.3675
iteration 45000     0.3738
iteration 50000     0.3930
iteration 55000     0.3799
iteration 60000     0.3602
iteration 65000     0.3745
iteration 70000     0.3816
iteration 75000     0.3752
iteration 80000     0.3582
iteration 85000     0.4025
iteration 90000     0.3873
iteration 95000     0.3761
iteration 100000    0.3729
iteration 105000    0.4246
iteration 110000    0.3850
iteration 115000    0.4084
iteration 120000    0.3983
```
I plot the loss by `tensorboard` (smoothed) and it looks strange.
![ethan_6006_ 3](https://user-images.githubusercontent.com/13479831/49021715-3cc68400-f1ce-11e8-8ade-3fb0273f5839.png)
![ethan_6006_ 4](https://user-images.githubusercontent.com/13479831/49021717-3cc68400-f1ce-11e8-9aa3-c84dee54f928.png)
![ethan_6006_ 5](https://user-images.githubusercontent.com/13479831/49021720-3cc68400-f1ce-11e8-8cbc-7fa4e627b55a.png)

@wasidennis 
Is this the case when you trained the model for the CVPR paper?

**Comment by wasidennis** [comment]
@EthanZhangYi thanks for your interest. Yes, I think your pytorch 0.4 version should be correct. Honestly, the instability issue is caused by the adversarial learning and the lack of ground truth labels in the target domain, so there is a difficulty to reach the convergence in the target domain via only using the annotations from source images. This should be the future direction that we need to put efforts on.
